src/web/app/pages-session/session-submission-page/session-submission-page.component.ts

```diff
@@ -341,8 +341,14 @@ this session.`;
         if (resp.status === 404) {
           this.simpleModalService.openInformationModal('Feedback Session Does Not Exist!', SimpleModalType.DANGER,
             'The session does not exist (most likely deleted by the instructor after the submission link was sent).');
-          this.navigationService.navigateWithErrorMessage(this.router, '/web/student/home', resp.error.message);
         }
+
+        if (resp.status === 403) {
+          this.simpleModalService.openInformationModal('Feedback Session Is Not Yet Open!', SimpleModalType.DANGER,
+            'The session is not yet open (the submission opening time is most likely updated by the instructor after the submission link was sent). <br/>You can access the session again once the session opens.');
```

Comment:
Is 403 _always_ equivalent to session not open? For all we know it could be an open session legitimately not visible to the user.

Reply:
Oh ya, there could be other cases also. Hmm, I think we need to change the messages.